### PR TITLE
Organize and show value types

### PR DESCRIPTION
**Pull request by kevinsimper**
I was a little confused about how `auto_reload` and what kind of types you had to provide. Hope this helps to clarify that for others.

Should it also say something about how `cache: true` and `auto_reload: true` works together, as that is a question i have myself :)

### CODE REVIEW

File: doc/api.rst [hunk 1]
```diff
@@ -72,29 +72,43 @@ options as the constructor second argument::
...
+  When set to ``true``, the generated templates have a
   ``__toString()`` method that you can use to display the generated nodes
   (default to ``false``).
 
-* ``charset``: The charset used by the templates (default to ``utf-8``).
+* ``charset`` *string (default to ``utf-8``)*
+
+  The charset used by the templates.
+
+* ``base_template_class`` *(default to ``Twig_Template``)*
```
Comment (sstok) [inline doc/api.rst]: maybe add a type here to?

File: doc/api.rst [hunk 2]
```diff
@@ -72,29 +72,43 @@ options as the constructor second argument::
...
+
+  The base template class to use for generated
+  templates.
 
-* ``base_template_class``: The base template class to use for generated
-  templates (default to ``Twig_Template``).
+* ``cache`` *boolean*
```
Comment (sstok) [inline doc/api.rst]: this can a boolean and string.

File: doc/api.rst [hunk 3]
```diff
@@ -72,29 +72,43 @@ options as the constructor second argument::
...
-* ``strict_variables``: If set to ``false``, Twig will silently ignore invalid
+* ``strict_variables`` *boolean*
+
+  If set to ``false``, Twig will silently ignore invalid
   variables (variables and or attributes/methods that do not exist) and
   replace them with a ``null`` value. When set to ``true``, Twig throws an
   exception instead (default to ``false``).
 
-* ``autoescape``: If set to ``true``, HTML auto-escaping will be enabled by
+* ``autoescape`` *boolean*
```
Comment (stof) [inline doc/api.rst]: this is not a boolean all the time

Comment (kevinsimper) [inline doc/api.rst]: Do you know what the values can be?

File: doc/api.rst (hunk 2 above)
Comment (kevinsimper) [inline doc/api.rst]: What is the string values?

Comment (stof) [inline doc/api.rst]: read the following sentence. It explains it. It is the path to the cache folder

File: doc/api.rst (hunk 3 above)
Comment (stof) [inline doc/api.rst]: one of the supported escaping strategies (``true`` is an alias for ``html`` for BC reasons)

File: doc/api.rst (hunk 2 above)
Comment (kevinsimper) [inline doc/api.rst]: So it should say `boolean/string`?

File: doc/api.rst (hunk 1 above)
Comment (SpacePossum) [inline doc/api.rst]: Maybe state that is the charset of the `output`  as well?
(as this is something I know others find confusing (ie. think it is charset of the data you use in Twig))

**Comment by SpacePossum** [comment]
nice PR :+1: , may I ask why you target master here? (1.x might benefit from this as well)

**Comment by kevinsimper** [comment]
Don't know if 1.x was there in februar? :)

File: doc/api.rst (hunk 2 above)
Comment (stof) [inline doc/api.rst]: ``string|false`` would be the actual one

File: doc/api.rst (hunk 3 above)
Comment (stof) [inline doc/api.rst]: btw, the following paragraphs are listing them

**Comment by fabpot** [comment]
Thank you @kevinsimper.

**Comment by fabpot** [comment]
I've fixed the remaining issues in d0b7130

**Pull request closed without merging**